### PR TITLE
Add test cases for controller

### PR DESCRIPTION
**Pull request by tkdchen**

### CODE REVIEW

File: controllers/component_image_controller_test.go [hunk 1]
```diff
@@ -435,6 +435,115 @@ var _ = Describe("Component image controller", func() {
...
+			setComponentAnnotationValue(resourceKey, ImageAnnotationName, string(repositoryInfoJsonBytes))
+			setComponentAnnotationValue(resourceKey, GenerateImageAnnotationName, `{"visibility": "public"}`)
+
+			Eventually(func() bool { return isCreateRepositoryInvoked }, timeout, interval).Should(BeTrue())
+
+			expectedValue, _ := json.Marshal(&ImageRepositoryStatus{Message: "failed to generate image repository"})
+			waitComponentAnnotationWithValue(resourceKey, ImageAnnotationName, string(expectedValue))
+		})
+
+		It("should stop and report error if fail to create an image repository", func() {
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply controllers/component_image_controller_test.go]: Yeah, removed. Thanks :)

File: controllers/component_image_controller_test.go [hunk 2]
```diff
@@ -435,6 +435,98 @@ var _ = Describe("Component image controller", func() {
...
+			waitComponentAnnotationGone(resourceKey, GenerateImageAnnotationName)
+			waitComponentAnnotation(resourceKey, ImageAnnotationName)
+
+			repoImageInfo := &ImageRepositoryStatus{}
+			component := getComponent(resourceKey)
+			Expect(json.Unmarshal([]byte(component.Annotations[ImageAnnotationName]), repoImageInfo)).To(Succeed())
+			Expect(repoImageInfo.Message).To(Equal("Invalid image url"))
+		})
+
+		It("should stop if fail to change repository visibility", func() {
```
Comment (mmorhun) [inline controllers/component_image_controller_test.go:475]: Looks like a duplicate of `should add message and stop if it's not possible to switch image repository visibility`

File: controllers/component_image_controller_test.go [hunk 3]
```diff
@@ -435,6 +435,98 @@ var _ = Describe("Component image controller", func() {
...
+			Eventually(func() bool { return isChangeRepositoryVisibilityInvoked }, timeout, interval).Should(BeTrue())
+
+			// Failed to change the visibility, reconciler should return immediately and annotations are not changed
+			waitComponentAnnotationUnchangedWithValue(testComponentKey, ImageAnnotationName, string(imageAnnotationValue))
+			waitComponentAnnotationUnchangedWithValue(testComponentKey, GenerateImageAnnotationName, generateAnnotationValue)
+
+			deleteComponent(testComponentKey)
+		})
+
+		It("should do nothing and set error if image annotation is invalid JSON", func() {
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply controllers/component_image_controller_test.go:511]: They are for `image.redhat.com/image` and `image.redhat.com/generate` individually.

File: controllers/component_image_controller_test.go (hunk 2 above)
Comment (tkdchen) [reply controllers/component_image_controller_test.go:475]: It covers this `else` branch https://app.codecov.io/gh/redhat-appstudio/image-controller/pull/55/blob/controllers/component_image_controller.go#L223

Comment (mmorhun) [reply controllers/component_image_controller_test.go:475]: Then the test name is incorrect, since it doesn't stop, but retries to change visibility.

**Review by mmorhun — APPROVED** [review]

**Pull request merged**